### DISCUSSION TITLE
is hbase suitable for meta engine?

### DISCUSSION THREAD

**Discussion by mxq-151** [juicedata/juicefs#3200] [General]
we have lots of small pics to store,pg,mysql,redis can not hold,and we think tikv is not reliable, we are familiar with hbase,so we want to make pr to juicefs by writing a hbase meta engine,is hbase suitable for meta engine?

**Comment by davies** [comment]
No, JuiceFS will issue lots of short query and expect response with low latency, so Hbase will NOT be a good option for meta engines. Also, JuiceFS require transactional API to make sure the atomic operations, HBase may not provide enough ACID guarantee.

**Reply by mxq-151** [reply]
> No, JuiceFS will issue lots of short query and expect response with low latency, so Hbase will NOT be a good option for meta engines. Also, JuiceFS require transactional API to make sure the atomic operations, HBase may not provide enough ACID guarantee.

1.HBase may not provide enough ACID guarantee,but phoenix provide,so juicefs can visit hbase by phoenix
2.the scenario is to write more, read less and update less
3.we found lots of short query in juicefs mknod, make Any committed changes in files can be visible on all servers immediately,so we want to remove Strong Consistency functionallity

is there other database you recommend？

**Reply by davies** [reply]
JuiceFS can use Redis or KeyDB, MySQL, PG, TiDB or TiKV as meta engine, do they work in your case?

**Reply by mxq-151** [reply]
> JuiceFS can use Redis or KeyDB, MySQL, PG, TiDB or TiKV as meta engine, do they work in your case?

we have considered TIKV before,but some user reported that TIKV is unstable

**Reply by davies** [reply]
We know that the biggest TiKV cluster for JuiceFS has 20+ billions files in it.

JuiceFS also support FoundationDB, it may scale better than TiKV, but we don't have much experience with it.

For Phoenix+HBase, we would worry that the latency will be too high.

**Reply by mxq-151** [reply]
> We know that the biggest TiKV cluster for JuiceFS has 20+ billions files in it.
> 
> JuiceFS also support FoundationDB, it may scale better than TiKV, but we don't have much experience with it.
> 
> For Phoenix+HBase, we would worry that the latency will be too high.

thanks for you advice
